app/services/file_attachment_upload_service.rb

```diff
@@ -40,4 +45,12 @@ def filename
     existing_filenames = revision.file_attachment_revisions.map(&:filename)
     UniqueFilenameService.new(existing_filenames).call(file.original_filename)
   end
+
+  def page_count
+    return unless file.content_type == PDF_CONTENT_TYPE
+
+    PDF::Reader.new(file.tempfile).page_count
+  rescue PDF::Reader::MalformedPDFError, PDF::Reader::UnsupportedFeatureError, OpenSSL::Cipher::CipherError
+    nil
+  end
```

Comment:
This looks good - does it seem like something we could test with a fixture?

Reply:
Sure, I figured we'd introduce the fixture when it's required for a test.  The File Attachment Upload Service doesn't have its own test file currently but I'd expect we'd introduce a test for page counts when we start displaying it in the app so the fixture can be added then.